packages/syft/src/syft/ast/klass.py

```diff
@@ -161,6 +172,93 @@ def run_class_method(
 
         return result
 
+    def run_class_smpc_method(
+        __self: Any,
+        *args: Tuple[Any, ...],
+        **kwargs: Any,
+    ) -> object:
+        """Run remote class method on a SharePointer and get pointer to returned object.
+
+        Args:
+            *args: Args list of class method.
+            **kwargs: Keyword args of class method.
+
+        Returns:
+            Pointer to object returned by class method.
+        """
+        # relative
+        from ..core.node.common.action.smpc_action_message import (
+            MAP_FUNC_TO_NR_GENERATOR_INVOKES,
+        )
+
+        seed_id_locations = kwargs.get("seed_id_locations", None)
+        if seed_id_locations:
+            raise ValueError(
+                "There should not be any kwargs named seed_id_locations in the kwargs for MPCTensor"
+            )
+
+        seed_id_locations = secrets.randbits(64)
+        kwargs["seed_id_locations"] = str(seed_id_locations)
+        generator = np.random.default_rng(seed_id_locations)
+
+        nr_ops = MAP_FUNC_TO_NR_GENERATOR_INVOKES[attr_path_and_name.split(".")[-1]]
+        for _ in range(nr_ops):
+            generator.bytes(16)
```

Comment:
Q:Why do we generate it sequentially without assignment of the generator.

Reply:
We use this to be "in sync" with the parties and the intermediary result from the shares.
All the parties (+ the orchestrator) have a seed and they generate "id_at_locations" in a deterministic and synchronized way.

The parties might have intermediary results (and they generate ids for those results).  The orchestrator discards the ids for those intermediary results and keeps only the last id for the result.